sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractModuleConfiguration.java

```diff
@@ -81,51 +85,51 @@ static String getAnalyzerReportDir(String languageKey) {
 
     // we don't generate sonar.cs.analyzer.projectOutPaths for test projects on purpose
     if (analyzerWorkDirPaths.isEmpty() && !configuration.hasKey("sonar.tests")) {
-      LOG.warn("Property missing: '" + getAnalyzerWorkDirProperty(languageKey) + "'. No protobuf files will be loaded for this project.");
+      LOG.warn("Property missing: '{}'. No protobuf files will be loaded for project '{}'.", getAnalyzerWorkDirProperty(languageKey), projectKey);
     }
 
     return analyzerWorkDirPaths.stream().map(x -> x.resolve(getAnalyzerReportDir(languageKey)))
-      .filter(AbstractModuleConfiguration::validateOutputDir)
+      .filter(dir -> validateOutputDir(dir, projectKey))
       .collect(Collectors.toList());
   }
 
-  private static boolean validateOutputDir(Path analyzerOutputDir) {
+  public List<Path> roslynReportPaths() {
+    String[] strPaths = configuration.getStringArray(getRoslynJsonReportPathProperty(languageKey));
+    LOG.debug("For project '{}', the Roslyn JSON report path has '{}'", projectKey, String.join(",", strPaths));
+    if (strPaths.length > 0) {
+      return Arrays.stream(strPaths)
+        .map(Paths::get)
+        .collect(Collectors.toList());
+    } else {
+      LOG.warn( "For project '{}', no Roslyn issues reports have been found.", projectKey);
+      return Collections.emptyList();
+    }
+  }
+
+  private static boolean validateOutputDir(Path analyzerOutputDir, String projectKey) {
```

Comment:
The passing of class state into static method seems painful and non-natural. Can you remove the `static` instead to simplify the call site to original form and the state management?

```suggestion
  private boolean validateOutputDir(Path analyzerOutputDir) {
```